plugins/nanospell/plugin.js

```diff
@@ -641,14 +641,16 @@
 
 			function getWords(block) {
 				var range = editor.createRange(),
+					currentWordObj,
 					words = [],
 					word;
 
 				range.selectNodeContents(block);
 				var wordwalker = new self.WordWalker(range);
 
-				while(word = wordwalker.getNextWord()) {
-					words.push(word.word);
+				while(currentWordObj = wordwalker.getNextWord()) {
+					word = currentWordObj.word;
+					if (word) words.push(word);
```

Comment:
Why does it work if we pull the if statement out to here?  There is a subtle difference, but it doesn't immediately click.  This would probably be worth some inline commenting.